### DISCUSSION TITLE
How to post formatted URLs?

### DISCUSSION THREAD

**Discussion by ShredderWalnut** [mastodon/mastodon#17337] [Q&A] [answered]
Hello! I noticed that in some account bios and toots, apparently URLs can be formatted as if using HTML. 

For example, [@Alamantus](https://cybre.space/@Alamantus)’s bio links to five of his Mastodon accounts on other instances, and all five links appear as `@Alamantus` instead of the full URL. I have also seen such formatted URLs appear in toots, although I cannot find an example now.

My question is, how can I post formatted URLs? Is this a custom feature available only on some instances? Thank you!

**Comment by Gargron** [comment, marked as answer]
Those are mentions. They autolink. The syntax is identical to the one in posts.

**Reply by ShredderWalnut** [reply]
Hi Eugen, thank you for the quick reply!

Sorry I didn’t make myself clear, but sometimes those formatted URLs I saw didn’t appear to be usernames, i.e. they didn’t even have an `@` at the beginning.

Luckily, I just happen to see such a toot on the federated timeline: [link to toot](https://mastodon.social/web/@lragir@%D5%AE%D5%B4%D5%A1%D5%AF%D5%B8%D6%82%D5%BF.%D5%B0%D5%A1%D5%B5/107654764668739254). I visited the account profile, and they have posted many toots with formatted URLs.

P.S.: Even though I translated the toot, I still had no idea what they were tooting about. I apologize if the example I posted is inappropriate in any ways, and will edit/delete this comment if necessary.

**Reply by Gargron** [reply]
That post originates from non-Mastodon software, they have different syntax for posts. Such a link is not possible to create in current Mastodon.

**Reply by ShredderWalnut** [reply]
I see. I should’ve realized this was the case, but I’m still not used to how awesome the Fediverse can be and assumed it *had* to be Mastodon software everywhere. 😆

**Reply by ShredderWalnut** [reply]
Hi @Gargron, sorry for another question :P

I just came across this post which I'm confident originates from a *Mastodon* instance. The post contains numerous formatted URLs pointing to Steam products:

https://rhabarberbarbara.bar/@unagi/107578901970468515

What's going on here? Thank you in advance!

**Reply by trwnh** [reply]
> I just came across this post which I'm confident originates from a Mastodon instance. The post contains numerous formatted URLs pointing to Steam products [...] What's going on here?

Most likely they merged custom code to allow it. The instance is heavily customized and its source code is available at https://github.com/rararwg/mastodon as seen in the footer.

**Reply by ShredderWalnut** [reply]
> The instance is heavily customized and its source code is available at https://github.com/rararwg/mastodon as seen in the footer.

I did not notice the footer. Thank you for your time and patience, @trwnh! My curiosity is satisfied. :P
